connect-example/src/main/java/com/stripe/android/connect/example/ui/embeddedcomponentmanagerloader/EmbeddedComponentLoaderViewModel.kt

```diff
@@ -33,27 +44,42 @@ class EmbeddedComponentLoaderViewModel @Inject constructor(
     val state: StateFlow<EmbeddedComponentManagerLoaderState> = _state.asStateFlow()
 
     init {
-        initializeManager()
+        loadManagerIfNecessary()
     }
 
-    // Public methods
-
     fun reload() {
         loadManager()
     }
 
-    // Private methods
+    override fun onCreate(owner: LifecycleOwner) {
+        val activity = owner as? ComponentActivity
+            ?: return
 
-    private fun initializeManager() {
-        val manager = embeddedComponentManagerProvider.provideEmbeddedComponentManager()
-        if (manager == null) {
-            loadManager()
-            return
+        // Bind appearance settings to the manager.
+        activity.lifecycleScope.launch {
```

Comment:
👀 Important that we bind in the Activity scope and not ViewModel scope, otherwise there'd be a memory leak